### PR TITLE
Lint, format and enable pre-commit checks

### PR DESCRIPTION
**Pull request by cgeorgii**
What it says in the title :)

### CODE REVIEW

File: fourmolu.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # # Number of spaces per indentation step
-# indentation: 4
+indentation: 2
```
Comment (cgeorgii) [inline fourmolu.yaml:2]: Note that this was 4 before but the entire codebase is formatted using 2, so I changed it to reflect the actual situation.

**Comment by seanhess** [comment]
I'm not a fan of pre-commit hooks: they get really annoying during rebases and other git deep dives. I'd rather enforce compliance using CI (github actions), and format on save with my editor. Github keeps changing their Haskell runner though, and I haven't figured out the new version. Do they have a nix runner? Could we use it to run tests? 

But I don't develop using nix, so I have no objection to a pre-commit hook there

**Comment by cgeorgii** [comment]
Hm, I never had an issue during rebase (other than having to manually run the hooks) but I get your pain. Since you don't use `nix`, I'll leave it enabled then.

As for CI, I can look into setting up a workflow with nix if that's cool with you.

**Comment by seanhess** [comment]
Sure, that would be great!

**Pull request merged**